packages/devtools_app/test/shared/server_api_client_test.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/shared/server_api_client.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() async {
+  group('computes the correct api URI', () {
+    const apiUriFor = DevToolsServerConnection.apiUriFor;
+    test('for a root URI without trailing slash', () async {
+      expect(
+        apiUriFor(Uri.parse('https://localhost:123?uri=x')),
+        Uri.parse('https://localhost:123/api/'),
+      );
+    });
+
+    test('for a root URI with trailing slash', () async {
+      expect(
+        apiUriFor(Uri.parse('https://localhost:123/?uri=x')),
+        Uri.parse('https://localhost:123/api/'),
+      );
+    });
+
+    test('for a /devtools/ URI with trailing slash', () async {
+      expect(
+        apiUriFor(Uri.parse('https://localhost:123/devtools/?uri=x')),
```

Comment:
`/devtools` is the path where DDS serves DevTools.

Reply:
For example:

```
$ xcodebuild/ReleaseX64/dart-sdk/bin/dart --observe --disable-service-auth-codes test_prog.dart 
The Dart VM service is listening on http://127.0.0.1:8181/
The Dart DevTools debugger and profiler is available at: http://127.0.0.1:8181/devtools/#/?uri=ws%3A%2F%2F127.0.0.1%3A8181%2Fws
```

Reply:
`/devtools/` works fine without any special casing, but I wasn't sure if there may be places that have `/devtools` without a trailing slash. Without the special case, resolving `api/` from `/devtools` would remove the devtools and I was worried that _if_ there were any places doing this it might break. It's possible there are none, or even that we'd serve a redirect to `/devtools/` but since I don't have visibility or the ability to test all places this is used I thought it was better to be sure it worked.

FWIW, the URL @bkonyi pasted is still has `/#` which should be removable now, though I'm not sure where that's being generated.